### ISSUE TITLE
modifies-value-receiver: incorrect vet on nested pointer struct

### ISSUE THREAD

**Issue by twmb** [mgechev/revive#636]
```go
package main

// B foo.
type B struct {
        V int
}

// C foo.
type C struct {
        *B
}

func (a C) foo() {
        a.V += 2
}

func main() {
        a := C{new(B)}
        a.foo()
}
```

```
junk.go:14:2: modifies-value-receiver: suspicious assignment to a by-value method receiver (revive)
	a.V += 2
```
a.V is modifying a pointer, not a value.

**Comment by chavacava** [comment]
Hi @twmb, thanks for reporting the bug.
The current implementation of the rule does not use type information thus it fails to identify `a.V` as a pointer.

**Comment by mgechev** [comment]
That's out of the scope of revive. There's a trade-off between type checking and performance and we decided to go with some false positives, but keep revive ⚡. I'll close for now, but will be happy to reconsider in case this turns out a popular request.
